resources/scripts/spack.sh

```diff
@@ -10,3 +10,37 @@ cd $SPACK_ROOT
 
 echo "export SPACK_ROOT=$SPACK_ROOT" >> /home/$USER/.bashrc
 echo "source \$SPACK_ROOT/share/spack/setup-env.sh" >> /home/$USER/.bashrc
+
+# Activate spack for some quick configuration
+source $SPACK_ROOT/share/spack/setup-env.sh
+
+# Find gcc and other compilers on PATH
+spack compiler find --scope site 
+spack external find --scope site
+
+# Root retains control of upstream packages
+cat ${SPACK_INSTALL_PREFIX}/spack/etc/spack/defaults/config.yaml > ~/.spack/config.yaml
+spack config --scope user add "config:module_roots:tcl:${SPACK_INSTALL_PREFIX}/spack/share/spack/modules"
+spack config --scope user add "config:module_roots:lmod:${SPACK_INSTALL_PREFIX}/spack/share/spack/lmod"
+
+# Users control packages local to their account
+spack config --scope site add "config:install_tree:root:~/.spack/install"
```

Comment:
Hey Evan, I want the spack packages to install in a shared dir, that's why I made the `$SPACK_ROOT` a user configurable parameter. Can we set these to `$SPACK_ROOT/.spack/install` instead of `~/.spack/install`?